src/matching_market.sol

```diff
@@ -45,17 +47,28 @@ contract MatchingMarket is MatchingEvents, ExpiringMarket, DSNote {
     mapping(address => uint) public _dust;                      //minimum sell amount for a token to avoid dust offers
     mapping(uint => uint) public _near;         //next unsorted offer id
     uint _head;                                 //first unsorted offer id
-    uint public dustId;                         // id of the latest offer marked as dust
 
+    // dust management
+    address DAI;
+    uint256 daiDustLimit;
+    UniswapSimplePriceOracle uniswapSimplePriceOracle;
 
-    constructor(uint64 close_time) ExpiringMarket(close_time) public {
+    constructor(address _DAI, uint256 _daiDustLimit, UniswapSimplePriceOracle _uniswapSimplePriceOracle) public {
+        DAI = _DAI;
```

Comment:
Maybe we could call it `baseToken` and `baseTokenDustLimit`. Something more generic.
Did you guys considered the possibility to compile this with solc 0.6.x so immutable can be used and save these storage loads?

Reply:
yes, makes sense I thought about it too.

As for `solc 0.6.x` I think we want to keep changes to the minimum. @maciejka thoughts?

Reply:
I am just not sure if `baseToken` is the best name... I am worried that it will be confusing with  base/quote of the market. WDYT?

Reply:
Yeah I agree `baseToken` is not the best name either. Maybe something else. Anyway I do not have strong feelings about it.

Reply:
What about `dustToken` and `dustTokenLimit`?